### PR TITLE
Add moinmoin

### PR DESCRIPTION
**Pull request by chadwcarlson**
Removed all files that could be downloaded during build. Updated README.

### CODE REVIEW

File: templates/moinmoin/files/.platform.app.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This file describes an application. You can have multiple applications
+# in the same project.
+#
+# See https://docs.platform.sh/user_guide/reference/platform-app-yaml.html
+
+# The name of this app. Must be unique within a project.
+name: app
+
+# The runtime the application uses.
+type: "python:2.7"
```
Comment (Crell) [inline templates/moinmoin/files/.platform.app.yaml]: Is it not Python 3 compatible?  What crummy software. :smile:

Comment (chadwcarlson) [reply templates/moinmoin/files/.platform.app.yaml]: Real pain is what it is.

File: templates/moinmoin/files/.platform.template.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  # Unique machine name, prefaced by a vendor or organization identifier
+  id: platformsh/moinmoin
+  # The human-readable name of the template.
+  name: MoinMoin
+  # Human-readable descriptive text for the template. Supports limited HTML.
+  description: MoinMoin is a wiki engine written for Python 2.7.
+  # A list of tags associated with the template.
+  tags:
+  - Python
+  - Language
```
Comment (Crell) [inline templates/moinmoin/files/.platform.template.yaml]: Shouldn't this be Python and "Wiki" or "CMS" or something like that?  "Language" sounds like a c&p goof.

File: templates/moinmoin/files/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Platform.sh MoinMoin (Python 2.7) minimal example
+
+This project provides a starter kit for hosting a MoinMoin wiki on Platform.sh. It is primarily an example, although it could be used as the starting point for a real project.
+
+MoinMoin is a wiki engine written for Python 2.7. This project is a fork of [MoinMoin 1.9](https://github.com/moinwiki/moin-1.9/tree/master/wiki) configured to run on Platform.sh. You can find more information about the MoinMoin project on [Github](https://github.com/moinwiki/moin-1.9/tree/master/wiki) or on their [website](https://moinmo.in/).
+
+All configuration files and page content are initialized here in `wiki`, but they are moved to the mounted directory `mywiki` during deployment so that the wiki engine is writable.
```
Comment (Crell) [inline templates/moinmoin/files/README.md]: "Wiki engine is writeable" - Danger Will Robinson!  The wiki engine itself needs to be writeable, not just a content dir?  Seriously?

File: templates/moinmoin/files/.platform.app.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+dependencies:
+    python:
+        pipenv: "2018.10.13"
+
+# The hooks executed at various points in the lifecycle of the application.
+hooks:
+    build: |
+        ./moin/scripts/install.sh
+    deploy: |
+        ./moin/scripts/config.sh
```
Comment (Crell) [inline templates/moinmoin/files/.platform.app.yaml]: Let's name these build.sh and deploy.sh so they're more self-documenting.

File: templates/moinmoin/files/moin/scripts/build.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+curl -O http://static.moinmo.in/files/moin-1.9.10.tar.gz
+tar xvzf moin-1.9.10.tar.gz
+
+mv seeds/wikiconfig.py moin-1.9.10
+mv seeds/wikiserverconfig.py moin-1.9.10
+
+cp -a moin-1.9.10/* /app/
+
+cd $PLATFORM_APP_DIR
```
Comment (Crell) [inline templates/moinmoin/files/moin/scripts/build.sh]: Nitpick: One blank line at the end of the file.

File: templates/moinmoin/files/moin/scripts/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+cd $PLATFORM_APP_DIR
+
+cp -a wiki/config mywiki
+cp -a wiki/server mywiki
+
+cp -a wiki/data/* mywiki/data
+cp -a wiki/underlay/* mywiki/underlay
+
+
+
```
Comment (Crell) [inline templates/moinmoin/files/moin/scripts/deploy.sh]: Nitpick: One blank line at the end of the file.

File: templates/moinmoin/files/.platform.app.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    build: |
+        ./setup/build.sh
+    deploy: |
+        ./setup/deploy.sh
+
+# The size of the persistent disk of the application (in MB).
+disk: 1024
+
+# Set a local R/W mount for logs
+mounts:
```
Comment (Crell) [inline templates/moinmoin/files/.platform.app.yaml]: The comment here seems out of date.

File: templates/moinmoin/files/setup/build.sh [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+# Stage MoinMoin log files
+mv ~/setup/editlog.py moin-1.9.10/MoinMoin/logfile
+mv ~/setup/eventlog.py moin-1.9.10/MoinMoin/logfile
+
+# Move the app directory
+cp -a ~/moin-1.9.10/* /app/
+
+# Remove the install
+rm -rf ~/moin-1.9.10
+rm ~/moin-1.9.10.tar.gz
```
Comment (Crell) [inline templates/moinmoin/files/setup/build.sh]: I don't know if we're going to be updating this all that often, but you can curl the file to a specific name.  Then you don't need to include the version number throughout the file.  That's a better practice in general, I think.

File: templates/moinmoin/files/setup/build.sh [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+curl -o moin-install.tar.gz http://static.moinmo.in/files/moin-$MOINVERSION.tar.gz
+tar xvzf moin-install.tar.gz
+mv moin-$MOINVERSION moin-install
+
+# Apply patches
+patch moin-install/MoinMoin/logfile/editlog.py ~/setup/editlog.patch
+patch moin-install/MoinMoin/logfile/eventlog.py ~/setup/eventlog.patch
+
+# Move install to app_dir
+cp -a ~/moin-install/* /app/
```
Comment (Crell) [inline templates/moinmoin/files/setup/build.sh]: Why is this a copy and not a move?

File: templates/moinmoin/files/setup/build.sh [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+# Remove the install
+rm -rf ~/moin-install
+rm ~/moin-install.tar.gz
+
+# Stage files that will go to mounts on deploy
```
Comment (Crell) [inline templates/moinmoin/files/setup/build.sh]: For clarify, edit comment that it's staging files that will get copied to mounts on *first* deploy.

Comment (Crell) [reply templates/moinmoin/files/setup/build.sh]: Also, nitpick: Comments should end in a period.

File: templates/moinmoin/files/setup/build.sh [hunk 11]
```diff
@@ -2,23 +2,23 @@
...
 patch moin-install/MoinMoin/logfile/eventlog.py ~/setup/eventlog.patch
 
-# Move install to app_dir
-cp -a ~/moin-install/* /app/
+# Move install to app_dir.
+mv ~/moin-install/* /app/
 
-# Remove the install
+# Remove the install.
 rm -rf ~/moin-install
```
Comment (Crell) [inline templates/moinmoin/files/setup/build.sh]: The move above makes this line no longer necessary.

**Pull request merged**